### PR TITLE
Add Llama-Cpp As Benchmark Option

### PR DESCRIPTION
**Pull request by zxybazh**
This PR introduces llama-cpp as one of the baselines in benchmarking script. Llama-cpp has it's own tokenize logic which is done on the c++ side.

Different versions of llama-cpp would impact the final results, currently working with the cublas enabled python binding of llama-cpp with version 0.1.50. Installation script is available at https://github.com/abetlen/llama-cpp-python#installation-with-openblas--cublas--clblast with the following script:
```
CMAKE_ARGS="-DLLAMA_CUBLAS=on" FORCE_CMAKE=1 pip install llama-cpp-python
```

### CODE REVIEW

File: tests/benchmark.py [hunk 1]
```diff
@@ -343,6 +355,94 @@ def generate(
...
+        self.tokens = self.model.tokenize("test".encode("utf-8"))
+        for i in range(2, max_gen_len * 3):
+            # we buffer a token list to avoid the overhead of calling tokenize() for each token
+            # first token is the start token, after that it's all "test" token
+            self.tokens.append(self.tokens[1])
+
+    def sync(self):
+        pass
+
+    def prep_model(self):
```
Comment (sunggg) [inline tests/benchmark.py:418]: Apology for the lack of the documentation, this `prep_model` is intended to prepare model with the clean kv cache. Can we use this?

**Review by sunggg — COMMENTED** [review]
Is this still a draft phase?

File: tests/benchmark.py (hunk 1 above)
Comment (zxybazh) [reply tests/benchmark.py:418]: Yeah sure will move to `prep_model`.

**Comment by zxybazh** [comment]
Thanks for checking back on this PR, this benchmark script can work now but the performance looks really bad so I'm checking details to see what can be impacting the performance. Currently at 2.35 tokens per second.

**Review by sunggg — APPROVED** [review]

**Pull request merged**